src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6675,38 +6739,174 @@ register_handler(int sig, struct sigaction *old_handler)
 
 /* Check whether current executable is sh/bash or not. Flag is_bash is set. */
-static void
+static inline void
 check_exe_sh_bash(void)
 {
+	if (memcmp(exe_short_name, "bash", 5) == 0 || memcmp(exe_short_name, "sh", 3) == 0)
+		is_bash = true;
+}
+
+#define CMDLINE_BUF_SIZE (2 * DFS_MAX_PATH + 2)
+
+static void
+extract_exe_name_1st_arg(void)
+{
+	FILE *f;
 	int   readsize;
-	/* the exe name extract from /proc/self/exe */
-	char *exe_path;
-	/* the short exe name from exe_path */
-	char *exe_short = NULL;
+	int   count = 0;
+	char *buf, *p, *end;
 
-	D_ALLOC(exe_path, DFS_MAX_PATH);
-	if (exe_path == NULL)
-		return;
+	f = fopen("/proc/self/cmdline", "r");
+	if (f == NULL) {
+		fprintf(stderr, "Fail to open file: /proc/self/cmdline. %d (%s)\n", errno,
+			strerror(errno));
+		exit(1);
+	}
+	buf = malloc(CMDLINE_BUF_SIZE);
+	if (buf == NULL) {
+		fprintf(stderr, "Fail to allocate memory for buf %d (%s)\n", errno,
+			strerror(errno));
+		exit(1);
+	}
+	readsize = fread(buf, 1, CMDLINE_BUF_SIZE, f);
+	if (readsize <= 0) {
+		fprintf(stderr, "Fail to read /proc/self/cmdline %d (%s)\n", errno,
+			strerror(errno));
+		fclose(f);
+		exit(1);
+	}
+
+	fclose(f);
 
-	readsize = readlink("/proc/self/exe", exe_path, DFS_MAX_PATH);
-	if (readsize >= DFS_MAX_PATH) {
-		DS_ERROR(ENAMETOOLONG, "path from readlink() is too long");
-		goto out;
-	} else if (readsize < 0) {
-		DS_ERROR(errno, "readlink() failed");
-		goto out;
+	exe_short_name = basename(buf);
+	if (exe_short_name == NULL) {
+		fprintf(stderr, "Fail to determine exe_short_name %d (%s)\n", errno,
+			strerror(errno));
+		exit(1);
+	}
+	exe_short_name = strndup(exe_short_name, DFS_MAX_NAME);
+	if (exe_short_name == NULL) {
+		printf("Fail to allocate exe_short_name %d (%s)\n", errno, strerror(errno));
+		exit(1);
+	}
+	first_arg = NULL;
+	end       = buf + readsize;
+
+	for (p = buf; p < end;) {
+		if (count == 1) {
+			if (p[0] == '/' || memcmp(p, "./", 2) == 0 || memcmp(p, "../", 3) == 0) {
+				/* Extract the first argument in command line */
+				first_arg = basename(p);
+				if (first_arg == NULL) {
+					fprintf(stderr, "Fail to determine first_arg %d (%s)\n",
+						errno, strerror(errno));
+					exit(1);
+				}
+				first_arg = strndup(first_arg, DFS_MAX_NAME);
+				if (first_arg == NULL) {
+					fprintf(stderr, "Fail to allocate first_arg %d (%s)\n",
+						errno, strerror(errno));
+					exit(1);
+				}
+			}
+			break;
+		}
+		count++;
+		while (*p++)
+			;
 	}
+	free(buf);
+	/* We allocated buffers for exe_short_name and first_arg. Now buf can be deallocated. */
+}
+
+static char *bypass_bash_cmd_list[] = {"autoconf", "configure", "libtool", "libtoolize",
+				       "lsb_release"};
 
-	exe_short = basename(exe_path);
-	if (exe_short == NULL) {
-		DS_ERROR(errno, "basename() failed");
-		goto out;
+static char *bypass_python3_cmd_list[] = {"scons", "scons-3", "dnf", "dnf-3", "meson"};
+
+static char *bypass_app_list[] = {"arch", "as", "awk", "basename", "bc", "cal", "cat", "chmod",
```

Comment:
as i mentioned this list should be documented better above.

Reply:
How much information we will provide about these applications? Would a list of them in document sufficient?

Reply:
IMO just list them above inthe md file where i a dded the bold comment

Reply:
ok. Thank you!

Reply:
The list is added in document. Thank you!